dev_tools/configure_vscode_for_bazel.py

```diff
@@ -239,41 +334,108 @@ def update_cc_build_file(bazel_patterns: list[str], bazel_args: list[str], confi
+        )
+        if success:
             logging.info("You can now run the debug target(s) in VS Code.")
-            recommended_actions.append(("bazel", "build", *args.additional_debug_arg, *args.bazel_pattern))
+            if not args.generate_build_targets:
+                recommended_actions.append(("bazel", "build", *args.additional_debug_arg, *args.bazel_pattern))
         else:
-            logging.error("No executable targets were found, no `launch.json` file was generated.")
+            logging.error("No executable targets found, no `launch.json` generated.")
+
+
+def handle_compile_commands_generation(
+    args: argparse.Namespace, vscode_dir: Path, recommended_actions: list[tuple[str, ...]]
+) -> None:
+    if args.generate_compile_commands:
+        success = update_cc_build_file(
+            args.bazel_pattern, args.additional_compile_commands_arg, vscode_dir / "BUILD.bazel", args.force
+        )
+        if success:
+            logging.info("You can now generate the `compile_commands.json` file.")
+            recommended_actions.extend(
+                [
+                    ("bazel", "build", *args.additional_compile_commands_arg, *args.bazel_pattern),
+                    ("bazel", "run", "//.vscode:refresh_compile_commands"),
+                ]
+            )
+
+
+def execute_recommended_actions(args: argparse.Namespace, recommended_actions: list[tuple[str, ...]]) -> None:
+    if recommended_actions:
+        if args.build:
+            for action in recommended_actions:
+                # action format is ("bazel", "command", "arg1", "arg2", ...)
+                run_bazel_command(*action[1:], verbose=args.verbose)
+        else:
+            # If not building immediately, provide suggestions to the user
+            suggested_cmds = "\n".join(" ".join(cmd) for cmd in recommended_actions)
+            logging.info("Remember to re-build the target(s) with:\n\n%s", suggested_cmds)
+
+
+def main() -> int:
+    args = parse_arguments()
+    configure_logging(verbose=args.verbose)
+
+    if not check_dependencies():
+        return 1
+
+    vscode_dir = get_workspace_root() / ".vscode"
+    setup_vscode_directory(vscode_dir)
 
-    if args.generate_compile_commands and update_cc_build_file(
-        args.bazel_pattern, args.additional_compile_commands_arg, vscode_dir / "BUILD.bazel", args.force
-    ):
-        logging.info("You can now generate the `compile_commands.json` file.")
-        recommended_actions.append(("bazel", "build", *args.additional_compile_commands_arg, *args.bazel_pattern))
-        recommended_actions.append(("bazel", "run", "//.vscode:refresh_compile_commands"))
+    executable_labels = generate_executable_labels(args)
 
-    if args.build:
-        for _bazel, *command in recommended_actions:
-            run_bazel_command(*command)
+    recommended_actions: list[tuple[str, ...]] = []
+    handle_tasks_json_generation(args, executable_labels, vscode_dir)
```

Comment:
I think we should view `task.json` and `lauch.json` generation as a unit. There is too much complexity by allowing all combinations.

Reply:
In my mind we should just always generate a `task.json` when we are asked to generate `lauch.json` as that is sort of a prerequisite.